overlord/snapshotstate/backend/reader.go

```diff
@@ -84,6 +84,15 @@ func Open(fn string) (reader *Reader, e error) {
 		return nil, err
 	}
 
+	// XXX: this mirrors the check from Iter()
```

Comment:
why the XXX bit of this here?  either we are happy?  or we add a setID argument to Open that we use if it's not 0, which seems easy enough if making the diff a bit larger?

leaving the XXX for the future seems not very useful here